AsyncThreadContextDefaultTest is unreliable

## Description

```
AsyncThreadContextDefaultTest.testAsyncLogWritesToLog:38->AbstractAsyncThreadContextTestBase.testAsyncLogWritesToLog:176 » ConditionTimeout Condition with Lambda expression in org.apache.logging.log4j.core.async.AbstractAsyncThreadContextTestBase was not fulfilled within 500 milliseconds.
```

## Configuration
**Version:** 2.x at 01125b11a7c52af12204a1c4a646158f63b2d6eb

**Operating system:** 
Microsoft Windows [Version 10.0.19045.4780

**JDK:** 
Apache Maven 3.9.9 (8e8579a9e76f7d015ee5ec7bfcdc97d260186937)
Maven home: C:\java\apache-maven-3.9.9
Java version: 17.0.12, vendor: Eclipse Adoptium, runtime: C:\Program Files\Eclipse Adoptium\jdk-17.0.12.7-hotspot
Default locale: en_US, platform encoding: UTF-8
OS name: "windows 10", version: "10.0", arch: "amd64", family: "windows"

## Logs

```
[ERROR] Tests run: 3, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 6.362 s <<< FAILURE! -- in org.apache.logging.log4j.core.async.AsyncThreadContextDefaultTest
[ERROR] org.apache.logging.log4j.core.async.AsyncThreadContextDefaultTest.testAsyncLogWritesToLog(Mode)[3] -- Time elapsed: 1.236 s <<< ERROR!
org.awaitility.core.ConditionTimeoutException: Condition with Lambda expression in org.apache.logging.log4j.core.async.AbstractAsyncThreadContextTestBase was not fulfilled within 500 milliseconds.
        at org.awaitility.core.ConditionAwaiter.await(ConditionAwaiter.java:167)
        at org.awaitility.core.CallableCondition.await(CallableCondition.java:78)
        at org.awaitility.core.CallableCondition.await(CallableCondition.java:26)
        at org.awaitility.core.ConditionFactory.until(ConditionFactory.java:1006)
        at org.awaitility.core.ConditionFactory.until(ConditionFactory.java:975)
        at org.apache.logging.log4j.core.async.AbstractAsyncThreadContextTestBase.testAsyncLogWritesToLog(AbstractAsyncThreadContextTestBase.java:176)
        at org.apache.logging.log4j.core.async.AsyncThreadContextDefaultTest.testAsyncLogWritesToLog(AsyncThreadContextDefaultTest.java:38)
        at java.base/java.lang.reflect.Method.invoke(Method.java:569)
        at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
        at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
        at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
        at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
        at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
        at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1845)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
        at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
        at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
        at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:276)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
        at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1625)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
        at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
        at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
        at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:276)
        at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1625)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
        at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
        at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
Caused by: java.util.concurrent.TimeoutException
        at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:204)
        at org.awaitility.core.Uninterruptibles.getUninterruptibly(Uninterruptibles.java:101)
        at org.awaitility.core.Uninterruptibles.getUninterruptibly(Uninterruptibles.java:81)
        at org.awaitility.core.ConditionAwaiter.await(ConditionAwaiter.java:103)
        ... 46 more
```

## Reproduction

```
mvn clean install
```

--- a/log4j-core-test/src/test/java/org/apache/logging/log4j/core/appender/SyslogAppenderTestBase.java
+++ b/log4j-core-test/src/test/java/org/apache/logging/log4j/core/appender/SyslogAppenderTestBase.java
@@ -16,6 +16,8 @@
  */
 package org.apache.logging.log4j.core.appender;
 
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static org.awaitility.Awaitility.await;
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertNotNull;
 import static org.junit.jupiter.api.Assertions.assertTrue;
@@ -49,7 +51,7 @@ abstract class SyslogAppenderTestBase {
             "TestApp - Audit [Transfer@18060 Amount=\"200.00\" FromAccount=\"123457\" ToAccount=\"123456\"]"
                     + "[RequestContext@18060 ipAddress=\"192.168.0.120\" loginId=\"JohnDoe\"] Transfer Complete";
     protected LoggerContext ctx = LoggerContext.getContext();
-    protected static final int DEFAULT_TIMEOUT_IN_MS = 100;
+    protected static final int DEFAULT_TIMEOUT_IN_MS = 3000;
     protected MockSyslogServer syslogServer;
     protected SyslogAppender appender;
     protected Logger root = ctx.getLogger("SyslogAppenderTest");
@@ -61,7 +63,7 @@ abstract class SyslogAppenderTestBase {
         LoggerContext.getContext().reconfigure();
     }
 
-    protected void sendAndCheckLegacyBsdMessages(final List<String> messagesToSend) throws InterruptedException {
+    protected void sendAndCheckLegacyBsdMessages(final List<String> messagesToSend) {
         for (final String message : messagesToSend) {
             sendDebugLegacyBsdMessage(message);
         }
@@ -69,7 +71,7 @@ abstract class SyslogAppenderTestBase {
         checkTheEqualityOfSentAndReceivedMessages(Level.DEBUG);
     }
 
-    protected void sendAndCheckLegacyBsdMessage(final String message) throws InterruptedException {
+    protected void sendAndCheckLegacyBsdMessage(final String message) {
         sendDebugLegacyBsdMessage(message);
         checkTheNumberOfSentAndReceivedMessages();
         checkTheEqualityOfSentAndReceivedMessages(Level.DEBUG);
@@ -80,7 +82,7 @@ abstract class SyslogAppenderTestBase {
         root.debug(message);
     }
 
-    protected void sendAndCheckStructuredMessages(final int numberOfMessages) throws InterruptedException {
+    protected void sendAndCheckStructuredMessages(final int numberOfMessages) {
         for (int i = 0; i < numberOfMessages; i++) {
             sendInfoStructuredMessage();
         }
@@ -88,7 +90,7 @@ abstract class SyslogAppenderTestBase {
         checkTheEqualityOfSentAndReceivedMessages(Level.INFO);
     }
 
-    protected void sendAndCheckStructuredMessage() throws InterruptedException {
+    protected void sendAndCheckStructuredMessage() {
         sendInfoStructuredMessage();
         checkTheNumberOfSentAndReceivedMessages();
         checkTheEqualityOfSentAndReceivedMessages(Level.INFO);
@@ -108,15 +110,13 @@ abstract class SyslogAppenderTestBase {
         root.info(MarkerManager.getMarker("EVENT"), msg);
     }
 
-    protected void checkTheNumberOfSentAndReceivedMessages() throws InterruptedException {
-        assertEquals(
-                sentMessages.size(),
-                getReceivedMessages(DEFAULT_TIMEOUT_IN_MS).size(),
-                "The number of received messages should be equal with the number of sent messages");
+    protected void checkTheNumberOfSentAndReceivedMessages() {
+        await().atMost(DEFAULT_TIMEOUT_IN_MS, MILLISECONDS)
+                .until(() -> syslogServer.getMessageList().size() == sentMessages.size());
     }
 
-    protected void checkTheEqualityOfSentAndReceivedMessages(final Level expectedLevel) throws InterruptedException {
-        final List<String> receivedMessages = getReceivedMessages(DEFAULT_TIMEOUT_IN_MS);
+    protected void checkTheEqualityOfSentAndReceivedMessages(final Level expectedLevel) {
+        final List<String> receivedMessages = getReceivedMessages();
 
         assertNotNull(receivedMessages, "No messages received");
         for (int i = 0; i < receivedMessages.size(); i++) {
@@ -148,10 +148,7 @@ abstract class SyslogAppenderTestBase {
         root.setLevel(Level.DEBUG);
     }
 
-    protected List<String> getReceivedMessages(final int timeOutInMs) throws InterruptedException {
-        synchronized (syslogServer) {
-            syslogServer.wait(timeOutInMs);
-        }
+    protected List<String> getReceivedMessages() {
         return syslogServer.getMessageList();
     }
 

--- a/log4j-core-test/src/test/java/org/apache/logging/log4j/core/async/AbstractAsyncThreadContextTestBase.java
+++ b/log4j-core-test/src/test/java/org/apache/logging/log4j/core/async/AbstractAsyncThreadContextTestBase.java
@@ -171,7 +171,7 @@ public abstract class AbstractAsyncThreadContextTestBase {
         for (int i = 0; i < LINE_COUNT; i++) {
             // buffer may be full
             if (i >= 128) {
-                waitAtMost(500, TimeUnit.MILLISECONDS)
+                waitAtMost(5, TimeUnit.SECONDS)
                         .pollDelay(10, TimeUnit.MILLISECONDS)
                         .until(() -> remainingCapacity.getAsLong() > 0);
             }

